src/adapters/mongoose.js

```diff
@@ -143,6 +143,17 @@ module.exports = (model, opts) => {
         // Schema
         return [schemaType(fieldInfo.options.type[0])];
       }
+
+      // NOTICE: Object with `type` reserved keyword.
+      //         See: https://mongoosejs.com/docs/schematypes.html#type-key).
+      if (fieldInfo.options.type[0] instanceof Object
+        && fieldInfo.options.type[0].type
+        // NOTICE: In case there is `[{type:{type:String}}]` which means "type" is used as property.
```

Comment:
I feel this comment can be misleading, because we talk about a case this bypassed according to the condition.
Here is my suggestion (maybe it's wrong):
```suggestion
        // NOTICE: Bypass for schemas like `[{ type: {type: String}, ... }]` where "type" is used as property, and thus we are in the case of an array of embedded documents.
```